doc/DOCUMENTATION.org

```diff
@@ -310,12 +313,13 @@ Where:
 | packages.el    | The list of packages to install and the functions to initialize them |
 
 =Packages= are =ELPA= packages which can be installed from an =ELPA= compliant
-repository, and =Extensions= are generally elisp code from git submodules.
+repository, and =Extensions= are generally elisp code from git submodules, or
+other code one might want to use but which isn't available on an =ELPA=
+compliant repository, for whatever reason.
```

Comment:
This section is out of date. I think we can remove it. The difference between packages will be described somewhere else.

The file tree of a layer must be updated. There is no `extensions.el` anymore and `extensions` folder is now `local`.

Although these changes are out of scope of this PR I'm OK to make them here.